contracts/truefi2/oracles/TimeAverageTruPriceOracle.sol

```diff
@@ -0,0 +1,104 @@
+// SPDX-License-Identifier: MIT
+pragma solidity 0.6.10;
+
+import {SafeMath} from "@openzeppelin/contracts/math/SafeMath.sol";
+import {AggregatorV3Interface} from "@chainlink/contracts/src/v0.6/interfaces/AggregatorV3Interface.sol";
+
+contract TimeAverageTruPriceOracle {
+    using SafeMath for uint256;
+
+    uint8 constant BUFFER_LEN = 255;
+    uint64 constant TIME_WINDOW = 7 days;
+
+    struct PricePoint {
+        uint256 runningTotal;
+        uint192 price;
+        uint64 timestamp;
+    }
+
+    PricePoint[BUFFER_LEN] public buffer;
+
+    AggregatorV3Interface public truPriceFeed;
+
+    uint256 public leftIndex;
+    uint256 public rightIndex;
```

Comment:
We don't want to lose storage in case of we need to upgrade the contract. Consider making the oracle `Upgradeable`.